pyracf/common/security_admin.py

```diff
@@ -258,7 +260,7 @@ def _build_bool_segment_dictionaries(self, segments: dict) -> None:
     def _build_segment_dictionaries(self, traits: dict) -> None:
         """Build segemnt dictionaries for each segment."""
         # Clear state for new request
-        self._clear_state()
+        self.__clear_state()
```

Comment:
I don't think this is necessary since state is cleared at the end of `_make_request`.

Reply:
It still was necessary, since there was no state clearing for `__generate_requests_only`, which the unit tests originally ran on. I changed this code

```python
        if self.__generate_requests_only:
            return self.__logger.redact_request_xml(
                security_request.dump_request_xml(encoding="utf-8"),
                secret_traits=self.__secret_traits,
            )
```
to the following.

```python
        if self.__generate_requests_only:
            request = self.__logger.redact_request_xml(
                security_request.dump_request_xml(encoding="utf-8"),
                secret_traits=self.__secret_traits,
            )
            self.__clear_state(clear_preserved_segment_traits=True)
            del security_request
            return request
```

This should happen anyway since we DO still want to clean up the inputs when we are generating a request only.